app/models/dashboard_presenter.rb

```diff
@@ -0,0 +1,9 @@
+class DashboardPresenter
+  attr_reader :solutions, :in_progress, :completed
+
+  def initialize(user)
```

Comment:
This will make the instructor dashboard so much smoother of a process - thanks for the reorganization!